lib/RGBot.js

```diff
@@ -583,196 +1098,360 @@ const RGBot = class {
-        if (block) {
-            if(skipCollection) {
-                // if we don't care about drops,
-                // then return true to indicate that the Block was broken
-                result = true
-            }
-            else
-            {
-                // if we care about drop then try to identify whether
-                // anything has dropped from breaking the Block
-                // and make the Bot move to collect it
-                try {
-                    if (await this.approachBlock(block)) {
-                        await this.digBlock(block);
-
-                        let droppedItem = null;
-                        await this.wait(25); // give the server time to create drops
-                        if(!block.drops || block.drops.length === 0) {
-                            droppedItem = this.findItemOnGround(block.name);
-                        }  else {
-                            const itemDefinition = this.getItemDefinitionById(block.drops[0].drop.id);
-                            if(!!itemDefinition) {
-                                const itemName = this.getEntityName(itemDefinition)
-                                droppedItem = !!itemName ? this.findItemOnGround(itemName) : null;
-                            }
-                        }
-                        if (droppedItem) {
-                            await this.approachEntity(droppedItem);
-                        }
-                        result = true;
+        const skipCollection = options.skipCollection || false
+        const maxDistance = options.maxDistance || 50
+        this.#logWarn(`findAndDigBlock(blockType, options): Deprecated since 1.2.0, please see doc example for new code`)
+        let candidateBlock = this.findBlocks({blockNames: [blockType], maxDistance: maxDistance}).shift()?.result
+        if (candidateBlock) {
+            if (await this.approachAndDigBlock(candidateBlock)) {
+                // you can collect the item here, or preferably, you can let collection be part of your main loop next pass decision process
+                if (!skipCollection) {
+                    await this.wait(5) // wait a few ticks for the item to spawn
+                    const theItem = this.findItemsOnGroundv2({itemNames: [blockType], maxDistance: 5}).shift()?.result
+                    if (theItem) {
+                        await this.collectItemOnGround(theItem)
                     }
                 }
-                catch (err) {
-                    console.error('Error finding and digging block', err)
-                }
+                return true
             }
         }
-        return result;
+        return false
+    }
+
+    /**
+     * Approach (path-find to) and dig the specified block.
+     * @param {Block} block
+     * @returns {Promise<boolean>}
+     */
+    async approachAndDigBlock(block) {
```

Comment:
We might need better naming on this and the above function. For a second I had to remember how `approachAndDigBlock` would be different from `findAndDigBlock`